omod/src/main/java/org/openmrs/module/coreapps/page/controller/MarkPatientDeadPageController.java

```diff
@@ -69,5 +84,16 @@ public String post(@SpringBean("patientService") PatientService patientService,
         }
         return conceptAnswers;
     }
+
+    /**
+     * This Method closes all active visits when a patient dies.
+     */
+    private void actionAfterDeath(Patient patient, Date deathDate) {
+        VisitService visitService = Context.getVisitService();
+        List<Visit> visitList = visitService.getActiveVisitsByPatient(patient);
+        for (Visit visit : visitList) {
```

Comment:
What happened if the visit list is `null` for a patient (not mentioned about empty list)?

Reply:
I put a a check to counter that